File: 0905-sort-array-by-parity/0905-sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        vector<int> ans(nums.size());
        int i = 0, j = nums.size() - 1, k = 0;
        while(k < nums.size()) {
            if(nums[k] % 2 == 0) {
                ans[i++] = nums[k++];
            }
            else {
                ans[j--] = nums[k++];
            }
        }
        return ans;

    }
};

File: 0905-sort-array-by-parity/0905-sort-array-by-parity.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function(nums) {
    let i = 0;

    for(let j = 0; j < nums.length; j++) {
        if(nums[j] % 2 === 0) {
            let temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
            i++;
        }
    }

    return nums;
};